src/cmap/conn/mod.rs

```diff
@@ -230,7 +235,7 @@ impl Connection {
         command: Command,
         request_id: impl Into<Option<i32>>,
     ) -> Result<CommandResponse> {
-        let message = Message::with_command(command, request_id.into());
+        let message = Message::with_command(command, request_id.into(), &self.server_api);
```

Comment:
Related to my comment above, I think the cleanest way to do this would be to have `Command` have a `set_api_version` method and then call that before passing the command to `Message::with_command`.

Reply:
Changed this. This allows me to simplify the logic in `Client::execute_operation_on_connection` and inject the client server API information in the command before emitting the CommandStarted event. On the other hand, this means that the connection will set the server API again, which wasn't in the original plan. I don't think this has any direct side effects, but wanted to point this out.

Reply:
I don't think the redundancy is a huge deal, but it might be mitigated if we do the change I mention above (i.e requiring that the API version is set before `send_command` rather than inside it).